internal/retry/config_test.go

```diff
@@ -0,0 +1,404 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package retry
+
+import (
+	"testing"
+	"time"
+
+	"github.com/crossdock/crossdock-go/assert"
+	"github.com/crossdock/crossdock-go/require"
+	"go.uber.org/yarpc/internal/backoff"
+	"go.uber.org/yarpc/internal/whitespace"
+	"gopkg.in/yaml.v2"
+)
+
+func TestConfig(t *testing.T) {
+	type testStruct struct {
+		msg string
+
+		retryConfig string
+
+		wantPolicyProvider *procedurePolicyProvider
+		wantError          []string
+	}
+	tests := []testStruct{
+		{
+			msg: "just default policy",
+			retryConfig: whitespace.Expand(`
+				policies:
+					once:
+						retries: 1
+						maxtimeout: 500ms
+						backoff:
+							exponential:
+								first: 10ms
+								max: 1s
+				default: once
+			`),
+			wantPolicyProvider: newPolicyProviderBuilder().registerDefault(
+				NewPolicy(
+					Retries(1),
+					MaxRequestTimeout(time.Millisecond*500),
```

Comment:
this is a unique case where we don't want testtime, we're only using this to assert against the time we got from the config (which won't be scaled)